Forta-Bots-Challenge-3/src/agent.ts

```diff
@@ -0,0 +1,69 @@
+import {
+  Finding,
+  HandleBlock,
+  getEthersProvider,
+  BlockEvent,
+  getAlerts,
+} from "forta-agent";
+import {
+  DAI_ADDRESS,
+  L1_ESCROW_ARBITRUM,
+  L1_ESCROW_FUNCTION_SIGNATURE,
+  L1_ESCROW_OPTIMISM,
+  BOT_ID,
+} from "./constants";
+import { ethers, Contract } from "ethers";
+import { getL1Finding, checkBlock } from "./utils";
+
+export function provideHandleBlock(
+  provider: ethers.providers.Provider,
+): HandleBlock {
+  return async function handleBlock(block: BlockEvent): Promise<Finding[]> {
+    const findings: Finding[] = [];
+    const { chainId } = await provider.getNetwork();
```

Comment:
Note that the SDK contains an `initialize` handler ([docs](https://docs.forta.network/en/latest/sdk/)) which you can use just like the `handleBlock` to initialize the chainId (will be executed only once).

Reply:
I referenced an example in the link you provided, let me know if the way I called it is correct in my latest commit, thanks!